### PR TITLE
docker: Add gfortran support

### LINKED ISSUE #1457: Could we add gfortran support to our images?
### Checklist

- [X] I've looked through the [issues and pull requests](https://github.com/cross-rs/cross/issues?q=) for similar request
- [X] This feature could be solved with a [custom docker image](https://github.com/cross-rs/cross#custom-docker-images) (optional)

### Describe your request

I'm working for a application with many math libs. And gfortran is always required.

It's not difficult to add it. For gnu, Add gfortran-${ARCH}-${SYSTEM}-gnu${ABI} is enough. And for musl, we can add `GCC_CONFIG += --enable-languages=c,c++,fortran` to `musl.sh` .

### Describe why this would be a good inclusion for `cross`

It's true that it will increase the size of the docker. But there's always gfortran in the cross-toolchain published by musl.cc or linaro. So I just think if we can add it to our image too. It will make cross docker more complete.

### PR DESCRIPTION
**Pull request by Dirreke**
Add gfortran support for all docker except target of `*-none*` and `*-andriod*`.

Close #1457

### CODE REVIEW

File: docker/dragonfly.sh [hunk 1]
```diff
@@ -95,7 +95,7 @@ EOF
         --disable-lto \
         --disable-multilib \
         --disable-nls \
-        --enable-languages=c,c++ \
+        --enable-languages=c,c++,,fortran \
```
Comment (Emilgardis) [inline docker/dragonfly.sh:98]:
```suggestion
        --enable-languages=c,c++,fortran \
```

File: docker/Dockerfile.powerpc64-unknown-linux-gnu [hunk 2]
```diff
@@ -14,6 +14,7 @@ FROM cross-base as build
 
 RUN apt-get update && apt-get install --assume-yes --no-install-recommends \
     g++-powerpc64-linux-gnu \
+    gfrotran-powerpc64-linux-gnu \
```
Comment (Emilgardis) [inline docker/Dockerfile.powerpc64-unknown-linux-gnu:17]:
```suggestion
    gfortran-powerpc64-linux-gnu \
```

File: docker/Dockerfile.mips-unknown-linux-gnu [hunk 3]
```diff
@@ -14,6 +14,7 @@ FROM cross-base as build
 
 RUN apt-get install --assume-yes --no-install-recommends \
     g++-mips-linux-gnu \
+    gfrotran-mips-linux-gnu \
```
Comment (Emilgardis) [inline docker/Dockerfile.mips-unknown-linux-gnu:17]:
```suggestion
    gfortran-mips-linux-gnu \
```

**Review by Emilgardis — CHANGES_REQUESTED** [review]
looks good so far!

One issue with this, it doesn't install it for the case where we use `Dockerfile.native`, which is used when doing `cargo build-docker-image x86_64-unknown-linux-gnu --platform="linux/x86_64=x86_64-unknown-linux-gnu"` (or just `cargo build-docker-image x86_64-unknown-linux-gnu`)

Easiest way to fix that would be to just install gfortran in linux-image.sh, perhaps add a install here: https://github.com/cross-rs/cross/blob/085092ca01d6ac9f5a0b612d9e68bcdf1e50f7ee/docker/linux-image.sh#L420-L423

**Comment by Emilgardis** [comment]
/ci try

lets try again!

**Comment by Dirreke** [comment]
Is any idea about why [aarch64-linux-android](https://github.com/cross-rs/cross/actions/runs/8398717508/job/23003943207?pr=1458#logs) failed? It seems that it just need more time to finish the build.

**Comment by Dirreke** [comment]
As [libc/3618](https://github.com/rust-lang/libc/pull/3618) has beed merged, let's try again @Emilgardis

**Comment by Dirreke** [comment]
It doesn't seem to be my fault. Should we try again? Or could we merge it?

**Review by Emilgardis — APPROVED** [review]
Lets merge this! Thank you!

**Comment by Dirreke** [comment]
timeout again. Could we adjust the settings of action?

**Comment by Emilgardis** [comment]
it should already be maxed out: https://github.com/cross-rs/cross/actions/runs/8907127039/workflow#L188 however I rememember something about it not applying like you would expect. Lets just try again

**Pull request merged**